fe/fe-core/src/main/java/com/starrocks/sql/analyzer/InsertAnalyzer.java

```diff
@@ -282,4 +288,23 @@ public static void analyze(InsertStmt insertStmt, ConnectContext session) {
         insertStmt.setTargetColumns(targetColumns);
         session.getDumpInfo().addTable(database.getFullName(), table);
     }
+
+    private static ExternalOlapTable getOLAPExternalTableMeta(Database database, ExternalOlapTable externalOlapTable) {
+        // copy the table, and release database lock when synchronize table meta
+        ExternalOlapTable copiedTable = new ExternalOlapTable();
+        externalOlapTable.copyOnlyForQuery(copiedTable);
+        boolean shouldLockAgain = false;
+        if (database.isReadLockHeldByCurrentThread()) {
+            database.readUnlock();
```

Comment:
since it's an reentrant lock, the read counter maybe greater that 1, after readUnlock(), the lock may be still held by current thread?

Reply:
done